### ISSUE TITLE
Audio desync between osu! and Audio Files

### ISSUE THREAD

**Issue by VINXIS** [ppy/osu-stable-issues#686]
On osu! cutting b20210311cuttingedge

Currently automatically offsets around 65 ms of audio to what 00:00:000 is meant to be

see https://osu.ppy.sh/beatmapsets/993059#osu/2076663 which has the first transient at around 58 ms in the audio file, but in editor the transient plays *behind* 00:00:000 which is causing offset issues for people playing (see https://osu.ppy.sh/beatmapsets/140944/discussion/-/generalAll#/2227100), once I added around 7 ms more of silence it seemed to be fixed for people I had test and for myself

Top is original Bottom is added silence
![image](https://i.imgur.com/XApYizO.png)

(Checked with Audio Compatibility **Off**)

**Comment by peppy** [comment]
i'm confused by this report. are you saying this is happening just for this one beatmap? are you saying this happens for all users or just some? does the same thing happen on lazer?

you say that "adding silence" fixes it, but are sure sure it's not just the reebcode? if there's an issue it's likely due to some weird mo3 metadata.

**Comment by VINXIS** [comment]
this is occurring at least for the 2 beatmaps i linked for sure, looking at the latest Ranked osu!Std beatmap https://osu.ppy.sh/beatmapsets/1378026#osu/2847746 the offset for the audio file's first transient seems like its around 25 ms from what i see

on lazer, just using the audio waveform the first timing point i place is around 12-13 ms https://i.imgur.com/f67EbTm.png although the audio is still starting a bit earlier than the waveform itself at least on my end and even shifting it by 1/8 using the 12-13 ms offset sounds better offset-wise to me at least even though that goes into the negative end again, showing in the .osu file that the first object is starting on `-28.411066735630214`. This is the osz i created in lazer https://cdn.discordapp.com/attachments/278271628587040769/822260713359278080/artist_-_title_VINXIS.osz

from what i saw at least looking at the reencode and the original mp3 used both for the 2 beatmaps linked shared the offsetting nature, both are constant 192 kbps bitrate mode, and LAME was used for the reencode; i can try with other mp3 encodings

Uhhh i dont know where 3 ms is coming from but for the original case it seemed like the offset was around 65 ms for the first transient to line up around 00:00:000 within precision, but i know about the 5 ms precision provided by windows tho

**Comment by peppy** [comment]
i'm still thoroughly confused by your reports. are you expecting audio to appear at 0ms? is this you wanting the game to match what you see in audacity?

**Comment by VINXIS** [comment]
Uh moreso the games matching visual indications like in lazer's internal waveform visualizer, not just audacity

**Comment by peppy** [comment]
This is reported on osu-stable-issues, did you put it in the wrong place?

**Comment by VINXIS** [comment]
I noticed it in stable initially which is when i made the issue, Im seeing this experience in both so im not sure

**Comment by peppy** [comment]
Can you make a video or post without using audacity explaining what the issue is? I'm again having huge trouble understanding what you're trying to get at here. The post you link to in the map is commenting on a 2-3ms offset and your latest post is talking about visual displays in lazer (known to be wrong due to the editor not properly applying platform offsets).

**Comment by VINXIS** [comment]
Oh, the comment on the beatmap discussion was from a player who wanted to suggest that the offset is off and may need looking into, after asking around it seemed to be more than just a  ~2-3 ms difference, with some people suggesting -10 or -15 even, which was when the first timing point was at around 00:00:001

Here's a video attempting to show what I'm trying to say is the issue: https://cdn.discordapp.com/attachments/515329811091030026/822274172772024320/2021-03-18_19-02-36.mp4 

The original difficulty has its offset as 00:00:000 but the notes are late w.r.t to the song, u can especially hear this at places where kick samples are used; with the -13 offset change at least from what I hear it sounds far more in line with the song, but the first object is now behind 00:00:000 and now at -00:00:013, heres an osz of the 2 difficulties https://cdn.discordapp.com/attachments/515329811091030026/822274706110939146/Demetori_-_Higan_Kikou__View_of_The_River_Styx.osz

**Comment by VINXIS** [comment]
id also say its a noticeable difference when playing the 2 different difficulties as well which i shouldve showed in the video

**Comment by peppy** [comment]
Right, that's a classic issue. Previously recommendations have been adding silence to the start of the track. I'll look into the particular beatmap when I have some time, but if you are in a hurry that would be the recommendation.

Additionally, if you want to experiment further, try compressing using a different MP3 compressor. Any kinds of delays which appear differently on audacity and in osu! can be found in metadata, where either of the applications if ignoring an offset that has been specified. I believe it is osu! in this case, as we intentionally do this to keep compatibility with older beatmaps, but YMMV.

**Comment by VINXIS** [comment]
Ye i added silence to the mp3 to deal with it for now and got the beatmap requalified, ill try with a few other compressors later in the week(s) to experiment further tho
